src/wp-admin/includes/image.php

```diff
@@ -384,30 +490,44 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
  * Errors are stored in the returned image metadata array.
  *
  * @since 5.3.0
+ * @since 6.1.0 The $mime_type parameter was added.
  * @access private
  *
- * @param array  $new_sizes     Array defining what sizes to create.
- * @param string $file          Full path to the image file.
- * @param array  $image_meta    The attachment meta data array.
- * @param int    $attachment_id Attachment ID to process.
+ * @param array  $new_sizes       Array defining what sizes to create.
+ * @param string $file            Full path to the image file.
+ * @param array  $image_meta      The attachment meta data array.
+ * @param int    $attachment_id   Attachment ID to process.
+ * @param string $mime_type       Optional. The mime type to check for missing sizes. Default is the image mime of $file.
+ * @param bool   $is_primary_mime Whether this subsize run is generating the primary mime type. Default true.
  * @return array The attachment meta data with updated `sizes` array. Includes an array of errors encountered while resizing.
  */
-function _wp_make_subsizes( $new_sizes, $file, $image_meta, $attachment_id ) {
+function _wp_make_subsizes( $new_sizes, $file, $image_meta, $attachment_id, $mime_type = '', $is_primary_mime = true ) {
```

Comment:
See my other comment, the `$is_primary_mime` is a bit weird to have as a parameter and based on the one usage that it has shouldn't be necessary.